nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java

```diff
@@ -255,13 +389,186 @@ public void throwExceptionIfNullOrEmptyNetworkElementList(List<NetworkElement> n
+
+  public PortChain createPortChain(PortChain portChain) {
+      checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+      portChain = portChain.toBuilder().id(null).build();
+      portChain = this.osClient.sfc().portchains().create(portChain);
+
+      return fixPortChainCollections(portChain);
+  }
+
+  public PortPairGroup createPortPairGroup(PortPairGroup portPairGroup) {
+      checkArgument(portPairGroup != null, "null passed for %s !", "Port Pair Group");
+      portPairGroup = portPairGroup.toBuilder().id(null).build();
+
+      portPairGroup = this.osClient.sfc().portpairgroups().create(portPairGroup);
+      return portPairGroup;
+  }
+
+  public PortPair createPortPair(PortPair portPair) {
+      checkArgument(portPair != null, "null passed for %s !", "Port Pair");
+      portPair = portPair.toBuilder().id(null).build();
+
+      portPair = this.osClient.sfc().portpairs().create(portPair);
+      return portPair;
+  }
+
+  public FlowClassifier getFlowClassifier(String flowClassifierId) {
+      return this.osClient.sfc().flowclassifiers().get(flowClassifierId);
+  }
+
+  public PortChain getPortChain(String portChainId) {
+      return fixPortChainCollections(this.osClient.sfc().portchains().get(portChainId));
+  }
+
+  public PortPairGroup getPortPairGroup(String portPairGroupId) {
+      return this.osClient.sfc().portpairgroups().get(portPairGroupId);
+  }
+
+  public PortPair getPortPair(String portPairId) {
+      return this.osClient.sfc().portpairs().get(portPairId);
+  }
+
+  public Port getPort(String portId) {
+      return this.osClient.networking().port().get(portId);
+  }
+
+  public FlowClassifier updateFlowClassifier(String flowClassifierId, FlowClassifier flowClassifier) {
+      checkArgument(flowClassifierId != null, "null passed for %s !", "Flow Classifier Id");
+      checkArgument(flowClassifier != null, "null passed for %s !", "Flow Classifier");
+
+      if (flowClassifier.getTenantId() != null) {
+          LOG.warn("Some attributes, (id, projectId) cannot be modified on %s. ", flowClassifier);
+      }
+      flowClassifier = flowClassifier.toBuilder().id(null).projectId(null).build();
+
+      return this.osClient.sfc().flowclassifiers().update(flowClassifierId, flowClassifier);
+  }
+
+  public PortChain updatePortChain(String portChainId, PortChain portChain) {
+      checkArgument(portChainId != null, "null passed for %s !", "Port Chain Id");
+      checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+
+      if (portChain.getTenantId() != null) {
+          LOG.warn("Some attributes,  (id, projectId) cannot be modified on %s. ", portChain);
+      }
+      portChain = portChain.toBuilder().id(null).projectId(null).build();
+
+      portChain = this.osClient.sfc().portchains().update(portChainId, portChain);
+      return fixPortChainCollections(portChain);
+  }
+
+  public PortPairGroup updatePortPairGroup(String portPairGroupId, PortPairGroup portPairGroup) {
```

Comment:
where is this called from? I see lot of check argument calls, but not sure if we are just being extra cautious by repeating or if really needed.

i dont see usage of updatePortPairGroup method in this change either...